### PR TITLE
[FIXED] Add a menu option to edit config file.

### PR DESCRIPTION
**Pull request by jwrb**
Should open the configuration file in the default text editor for the machine.
Note this has only been tested on Mac. Should work on Linux
Addresses #328 and replaces PR [#399](https://github.com/PaycoinFoundation/paycoin/pull/399)

### CODE REVIEW

File: src/qt/bitcoingui.cpp [hunk 1]
```diff
@@ -950,3 +955,14 @@ void BitcoinGUI::toggleHidden()
...
+
+void BitcoinGUI::editConfig()
+{
+    QMessageBox mb;
+    mb.setWindowTitle(tr("Notice"));
+    mb.setText(tr("Any changes made to the configuration will require restarting the application."));
+    mb.exec();
+    QString configFile=QString::fromStdString(GetConfigFile().string());
+    //QDesktopServices::openUrl(QUrl(QString::fromStdString(GetConfigFile().string())));
+    QDesktopServices::openUrl(QUrl::fromLocalFile(configFile));
```
Comment (IngCr3at1on) [inline src/qt/bitcoingui.cpp]: Can this be changed back to avoid the need for the extra variable? We don't need the commented line either :D

**Comment by jwrb** [comment]
I'll have a look @IngCr3at1on. It should work now. :)

**Comment by ligerzero459** [comment]
Open in my default editor on OS X :+1:

**Comment by jwrb** [comment]
Awesome... as soon as @IngCr3at1on can confirm it's working on for him we'll be good!

**Comment by ligerzero459** [comment]
I've got time. I'll boot up my Linux VM and give it a quick test

**Comment by jwrb** [comment]
Ah great! thanks!

**Comment by ligerzero459** [comment]
Working like a charm on Linux as well. LGTM

**Comment by IngCr3at1on** [comment]
I thought I commented already yes this works for me. Did anyone test it on windows yet?

**Comment by jwrb** [comment]
I'll fire up a windows VM.

**Comment by mitchellcash** [comment]
I've tested this on Windows and it works. The only nit I have is that the file type `.conf` is not used by any default application on Windows. So on first run you are presented with:



If from here you select an application from the list of installed programs and force it as your future default well then the feature works as expected.

@jwrb would you mind checking out #420.

**Comment by IngCr3at1on** [comment]
@mitchellcash yeah the file association thing is really expected. Not sure there's anything we can do about that.

**Comment by mitchellcash** [comment]
@IngCr3at1on I agree. I started to write down thoughts and comments but then sort of realised I had no real solution for it.

I think the benefits in this one were worth pulling into the codebase.

**Pull request merged**